de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/listeners/OpenFileListener.java

```diff
@@ -1,34 +1,56 @@
-package de.tu_bs.cs.isf.e4cf.parts.project_explorer.listeners;
-
-import java.util.Map;
-
-import javax.inject.Inject;
-
-import org.eclipse.jface.viewers.DoubleClickEvent;
-import org.eclipse.jface.viewers.IDoubleClickListener;
-
-import de.tu_bs.cs.isf.e4cf.core.file_structure.FileTreeElement;
-import de.tu_bs.cs.isf.e4cf.core.util.ServiceContainer;
-import de.tu_bs.cs.isf.e4cf.parts.project_explorer.interfaces.IProjectExplorerExtension;
-
-public class OpenFileListener implements IDoubleClickListener {
-	@Inject private ServiceContainer container;
-	private Map<String,IProjectExplorerExtension> fileExtensions;
-	
-	public OpenFileListener(Map<String, IProjectExplorerExtension> fileExtensions) {
-		this.fileExtensions = fileExtensions;
-	}
-	
-	@Override
-	public void doubleClick(DoubleClickEvent event) {	
-		FileTreeElement target = container.rcpSelectionService.getCurrentSelectionFromExplorer();
-		if(target != null && !target.isDirectory()) {
-			String path = target.getAbsolutePath();
-			String fileExtension = path.substring(path.lastIndexOf(".")+1);
-			//if a double click action is available for given extension execute it.
-			if(fileExtensions.containsKey(fileExtension)) {
-				fileExtensions.get(fileExtension).execute(container);
-			}
-		}
-	}
-}
+package de.tu_bs.cs.isf.e4cf.parts.project_explorer.listeners;
+
+import java.util.Map;
+
+import org.eclipse.e4.core.contexts.IEclipseContext;
+import de.tu_bs.cs.isf.e4cf.core.file_structure.FileTreeElement;
+import de.tu_bs.cs.isf.e4cf.core.util.ServiceContainer;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.interfaces.IProjectExplorerExtension;
+import javafx.event.EventHandler;
+import javafx.scene.input.MouseButton;
+import javafx.scene.input.MouseEvent;
+
+
+public class OpenFileListener implements EventHandler<MouseEvent> {
+	
+	IEclipseContext eclipseContext;
+	private ServiceContainer services;
+	private Map<String,IProjectExplorerExtension> fileExtensions;
+	
+	public OpenFileListener(IEclipseContext eclipseContext, Map<String,
+			IProjectExplorerExtension> fileExtensions,
+			ServiceContainer services) {
+		this.eclipseContext = eclipseContext;
+		this.fileExtensions = fileExtensions;
+		this.services = services;
+	}
+	
+	/**
+	 * Add double-click listener to the files of the project explorer
+	 * 
+	 * @param eclipseContext
+	 */
+	@Override
+	public void handle(MouseEvent event) {
+		// left mouse button and double click required
+		if(event.getButton().equals(MouseButton.PRIMARY) && event.getClickCount() == 2) {
+			// should not trigger on directories
+			FileTreeElement target = services.rcpSelectionService.getCurrentSelectionFromExplorer();
+			if(!target.isDirectory()) {
+				openFile(target);
+			}
+		}
+	}
+	
+	private void openFile(FileTreeElement target) {
+		if(target != null) {
+			String path = target.getAbsolutePath();
+			String fileExtension = path.substring(path.lastIndexOf(".")+1);

```

Comment:
`getExtension()` is provided by FileTreeElement

Reply:
improvement has been applied